demo/addons/godot-openxr/scenes/first_person_controller_vr.gd

```diff
@@ -1,27 +1,46 @@
 extends ARVROrigin
 
+signal initialised
+signal failed_initialisation
+
+export var auto_initialise = true
+export var start_passthrough = false
 export (NodePath) var viewport = null
 
 var interface : ARVRInterface
 
+func _ready():
+	if auto_initialise:
+		initialise()
+
 func initialise() -> bool:
-	var interface = ARVRServer.find_interface("OpenXR")
+	if interface:
+		# we are already initialised
+		return true
+
+	interface = ARVRServer.find_interface("OpenXR")
 	if interface and interface.initialize():
 		print("OpenXR Interface initialized")
 
-		# Comment to disable passthrough
-		$Configuration.start_passthrough()
-
-		# Connect to our plugin signals
-		_connect_plugin_signals()
-
+		# Find the viewport we're using to render our XR output
 		var vp : Viewport = null
 		if viewport:
 			vp = get_node(viewport)
 
 		if !vp:
 			vp = get_viewport()
 
+		# Start passthrough?
+		if start_passthrough:
+			# make sure our viewports background is transparent
```

Comment:
@m4gr3d I've also added an export variable to enable/disable pass through. I don't think it's a good idea to suggest to the user to make changes in the files that are in the plugin. This means that whenever they update the plugin with a newer version, they need to make the changes again. With export variables the decision is recorded in their own project and it becomes easier to upgrade to newer versions.

Seeing the transparent background is a little double here but as we know the correct viewport here it seemed like a smart thing to do.